lib/devise/passkeys/controllers/passkeys_controller_concern.rb

```diff
@@ -94,11 +93,14 @@ def user_details_for_registration
         end
 
         def verify_passkey_challenge
-          if parsed_credential.nil?
-            render json: { message: find_message(:credential_missing_or_could_not_be_parsed) }, status: :bad_request
-            delete_registration_challenge
-            return false
+          begin
+            if parsed_credential.nil?
+              return render_credential_missing_or_could_not_be_parsed_error
+            end
+          rescue JSON::JSONError, TypeError
+            return render_credential_missing_or_could_not_be_parsed_error
           end
```

Comment:
love the error handling, but how about:
```ruby
before_action :verify_credential_integrity

def verify_credential_integrity
  return render_credential_missing_or_could_not_be_parsed_error if parsed_credential.nil?
rescue JSON::JSONError, TypeError
  return render_credential_missing_or_could_not_be_parsed_error
end
```

these two methods clearly have separate responsibilities, and as a bonus we get rid of all the `begin`s

Reply:
Great idea, implementing!

Reply:
Implemented in https://github.com/ruby-passkeys/devise-passkeys/pull/29/commits/f1400cb4b217c20b9e74fda3f55f74284e373d25